examples/stress_tests/many_components.rs

```diff
@@ -124,23 +127,40 @@ fn stress_test(num_entities: u32, num_components: u32, num_systems: u32) {
     // spawn a bunch of entities
     for _ in 1..=num_entities {
         let num_components = rng.gen_range(1..10);
-        let components = component_ids.choose_multiple(&mut rng, num_components);
+        let components: Vec<ComponentId> = component_ids
+            .choose_multiple(&mut rng, num_components)
+            .copied()
+            .collect();
 
         let mut entity = world.spawn_empty();
-        for &component_id in components {
-            let value: u8 = rng.gen_range(0..255);
-            OwningPtr::make(value, |ptr| {
+        let values: Vec<OwningPtr> = components
+            .iter()
+            .map(|_id| {
+                let mut value: u8 = rng.gen_range(0..255);
+                let value = &mut value;
+                // SAFETY: value is non null since it was initialized from an &mut above
+                #[allow(unsafe_code)]
+                let ptr = unsafe { NonNull::new_unchecked(value as *mut u8) };
                 #[allow(unsafe_code)]
-                // SAFETY:
-                // component_id is from the same world
-                // value is u8, so ptr is a valid reference for component_id
                 unsafe {
-                    entity.insert_by_id(component_id, ptr);
+                    OwningPtr::new(ptr)
```

Comment:
Isn't `ptr` pointing to a variable local to the closure?  I think this will be dangling by the time it gets passed to `insert_by_ids`.  You might need to create a `Vec<u8>` with values so that they can all be alive at once.

Reply:
Whoops, you're right. Surprised this doesn't blow up.

Reply:
Fixed it by storing it as a `Vec<ManuallyDrop<u8>>`.  The extra allocation slowed things down from 10s -> 15s.

Reply:
If the allocation is the slow part, you might be able to get some perf back by re-using the `Vec`.  Declare it outside the loop, `extend()` to populate it, and `drain()` instead of `into_iter()` to consume it.  

I think `ManuallyDrop<u8>` is overkill since `u8` is already `Copy`, but it's harmless at worst.  

It might be a little safer to do `PtrMut::from(value).promote()` so that the compiler can check the lifetimes, but I'm pretty sure what you have is sound.

Reply:
ah nice, I didn't know about that way of constructing a OwningPtr